.github/workflows/app.yml

```diff
@@ -0,0 +1,36 @@
+name: Test and build Jellyfin Android
+on:
+  push:
+  release:
+    types: [ published ]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Cache Gradle wrapper and dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/wrapper/dists
+            ~/.gradle/caches
+          key: gradle-cache
```

Comment:
Any particular reason youu are not using the "default" `key` and `restore-key` for a Gradle build?

```yaml
- uses: actions/cache@v2
  with:
    path: |
      ~/.gradle/caches
      ~/.gradle/wrapper
    key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
    restore-keys: |
      ${{ runner.os }}-gradle-
```

Reply:
The reason I am asking is that the key you selected may work but you run into the risk of hitting the [Cache Limit](https://github.com/actions/cache#cache-limits) somewhere down the line.

Reply:
We should follow the guide for gradle for the caching: https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-java-with-gradle#caching-dependencies

Reply:
Reworked the caching mechanism now.